Yoast/wordpress-seo
# Changed the amount of images queried to be placed in OG tags: now exits after first suitable image found.

## Summary

This PR can be summarized in the following changelog entry:

* Optimises the query selecting a fallback OG image for better performance.  

## Relevant technical choices:

* Instead of creating a query to the database for each image on the page and adding all of these images to the og tags, the method now exits after the first image that meets the requirements is found. 
* A new function for caching attachment ID's is introduced in the image utils class. 

## Test instructions

This PR can be tested by following these steps:

* Create a page with several images. 
* Make sure the page does not have a featured image.
* Make sure this page does not have any social media images configured.
* Open the page/post and right click to open "view page source", you should see a `og:image` tag  for each image on the page. 
* Switch to this branch and do the same: there should be only one `og:image` tag now.  

## Quality assurance


Fixes #9679
